src/posthog-core.ts

```diff
@@ -2152,9 +2154,34 @@ export class PostHog {
     }
 
     _captureInitialPageview(): void {
-        if (document && !this._initialPageviewCaptured) {
+        if (!document) {
+            return
+        }
+
+        // If page is not visible, add a listener to detect when the page becomes visible
+        // and trigger the pageview only then
+        // This is useful to avoid `prerender` calls from Chrome/Wordpress/SPAs
+        // that are not visible to the user
+
+        if (document.visibilityState !== 'visible') {
+            if (!this._visibilityStateListener) {
+                this._visibilityStateListener = this._captureInitialPageview.bind(this)
+                addEventListener(document, 'visibilitychange', this._visibilityStateListener)
+            }
```

Comment:
what if you do this during the SDK init instead? so the listener is ready once `_captureInitialPageview` is called

Reply:
Hm, yeah. I thought adding this here made it simpler and simplified the main function's concern, but now that I need this flag it might make sense to put it back there, I'll reconsider

Reply:
Ok, I cant do this because there are two places that call this function: either on `init` or when calls `opt_in_capturing` so it makes sense to keep this here.